docs/client.rst

```diff
@@ -72,18 +72,111 @@ WBEM Operation recording
+:method:`~pywbem.enable` method.
+
+In addition, the logger names for the operations and http loggers should be
+defined to create usable loggers since the pywbem loggers use attributes in
+addition to the standard python loggers.  This is done with the
+:class:`~pywbem.PywbemLoggers` and specifically with the method
+:method:`~pywbem.PywbemLoggers.create_logger` which creates one or more loggers
+from the input parameters.
+
+::
+
+    logger.setLevel(logging.DEBUG)
+    # To add the LogOperationRecorder
+    # Enable the logs using one of the methods in PywbemLoggers
+    PywbemLoggers.create_logger('all')   # enables both loggers
+    # Create the connection and enable the logger
+    conn = WBEMConnection(...)
+    logging = LogOperationRecorder()
+    add_operations_recorder(LogOperationRecorder())
+    # The LogOperationRecorder will not be active.
+    # To enable writing logs
+
+
+    # To add both recorders
+    logger.setLevel(logging.DEBUG)
+    PywbemLoggers.create_logger('all')   # enables both loggers
+    conn = WBEMConnection(...)
+    add_operations_recorder(LogOperationRecorder())
+    yamlfp = TestClientRecorder.open_file(self.yamlfile, 'a')
+    add_operations_recorder(TestClientRecorder(yamlfp))
+
+    # Both TestClientRecorder and LogOperationRecorder are be
+    # active, enabled and recording
+    # To change the enabled state of either, use the enabled/disabled
+    # methods of the Recorder
+
+    logging.disable()   # disables recording to the log
+
+Activated loggers can computing-wise expensive so it is best not to activate
+either logger unless they are to be used for that specific WBEMConnection.
+
+The Recorder enable() and disable() methods simply set flags to bypass creating
+the final recorder output so activating and disabling is still more
+compute-wise expensive than not activating a recorder at all.
+
+The :class:`~pywbem.LogOperationRecorder' is a more complete mechanism than the
+:attr:`~pywbem.WBEMConnection.debug` in that it records
+information on operations and http interactions with a WBEM server to a python
+log file.  This information includes the following:
+
+a. If operations logging is set (the log component == 'ops':
+    1. Records the method input parameters
+    2. Records either all or a fixed length of the respons values with an
+       optional maximum length attribute because this log entry can become
+       enormous if it records complete responses.
+    3. Records all exceptions
+
+b. If http logging is set (the log component == 'http':
+    1. Records the HTTP attributes and data of the request
+    2. Records the HTTP attributes and data of the response with an optional
+       max length
+c.  If either 'http' or 'ops' log set it records the WBEMConnection attributes
+    and an id so that ops and http logs can be linked back to the WBEM
+    connection.
+
+d. If the log component = 'all' both operations and http are logged.
+
+Logs may be recorded to 'stderr' or a 'file'.
```

Comment:
Why do you put "stderr" and "file" in quotes? This is just a normal sentence, and should not give hidden hints about a syntax that is not really explained.

Reply:
Changed